linkerd/install.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2019 Layer5.io
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linkerd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	repoURL              = "https://api.github.com/repos/linkerd/linkerd2/releases/latest"
+	emojivotoInstallFile = "https://run.linkerd.io/emojivoto.yml" // switching to a template because namespace was hardcoded
+	booksAppInstallFile  = "https://run.linkerd.io/booksapp.yml"
+
+	cachePeriod = 1 * time.Hour
+)
+
+var (
+	URLSuffix = "-" + runtime.GOOS
+	localFile = path.Join(os.TempDir(), "linkerd-cli")
+	// emojivotoLocalFile = path.Join(os.TempDir(), "emojivoto.yml")
+	booksAppLocalFile = path.Join(os.TempDir(), "booksapp.yml")
+)
+
+type APIInfo struct {
+	TagName    string   `json:"tag_name,omitempty"`
+	PreRelease bool     `json:"prerelease,omitempty"`
+	Assets     []*Asset `json:"assets,omitempty"`
+}
+
+type Asset struct {
+	Name        string `json:"name,omitempty"`
+	State       string `json:"state,omitempty"`
+	DownloadURL string `json:"browser_download_url,omitempty"`
+}
+
+func (iClient *LinkerdClient) getLatestReleaseURL() error {
```

Comment:
You might want to use https://versioncheck.linkerd.io/version.json instead. That'll help differentiate between stable and edge releases as well.